File: Wirtualny Swiat/PiotrPrzymus193736JavaWS/Guarana.java
package WSJAVA;
import java.util.Random;

public class Guarana extends Roslina{
    public Guarana(int polX, int polY, Swiat swiat) {
        super(polX, polY, swiat);
        this.sila = 0;
        this.inicjatywa = 0;
        this.znak_gatunku = Stale.GUARANA;
        this.wiek = 0;
        this.nazwa = "Guarana";
    }
    public Guarana(int polozenieX, int polozenieY, int sila, int inicjatywa, char znak_gatunku, int wiek, String nazwa, Swiat swiat) {
        super(polozenieX, polozenieY, sila, inicjatywa, znak_gatunku, wiek, nazwa, swiat);
    }

    @Override
    public void kolizja(Organizm o)
    {
        o.SetSila(o.GetSila() + 3);
        String px = String.valueOf(polozeniex);
        String py = String.valueOf(polozeniey);
        String p2x = String.valueOf(o.GetPolozenieX());
        String p2y = String.valueOf(o.GetPolozenieY());
        //System.out.println(o.GetNazwa() + "(" + p2x + "," + p2y + ") zjadl " + nazwa + "(" + px + "," + py + "), dostaje +3 sily");
        swiat.WypiszZdarzenie(o.GetNazwa() + "(" + p2x + "," + p2y + ") zjadl " + nazwa + "(" + px + "," + py + "), dostaje +3 sily");

        swiat.UsunOrganizm(polozeniex, polozeniey);
    }

    @Override
    public void PostawDziecko(int x, int y)
    {
        Random rand = new Random();
        int szansa = rand.nextInt(Stale.RNG_GUARANA);
        if (szansa == 0)
        {
            //System.out.println(nazwa + "(" + polozeniex + "," + polozeniey + ") rozsiewa sie na (" + x + "," + y + ")");
            swiat.WypiszZdarzenie(nazwa + "(" + polozeniex + "," + polozeniey + ") rozsiewa sie na (" + x + "," + y + ")");
            swiat.DodajOrganizm(new Guarana(x, y, swiat));
            //swiat.gotoxy(40, 8);
            return;
        }
        //System.out.println(nazwa + "(" + polozeniex + "," + polozeniey + ") nie rozsiewa sie");
        swiat.WypiszZdarzenie(nazwa + "(" + polozeniex + "," + polozeniey + ") nie rozsiewa sie");
        //swiat.gotoxy(40, 8);
    }
}

File: Wirtualny Swiat/Piotr Przymus 193736 Wirtualny Swiat/Guarana.cpp
#include "Guarana.h"

Guarana::Guarana(int polX, int polY, Swiat& swiat)
	: Roslina(polX, polY, swiat)
{
	wiek = 0;
	inicjatywa = 0;
	sila = 0;
	znak_gatunku = GUARANA;
	nazwa = "Guarana";
}

Guarana::Guarana(int polozenieX, int polozenieY, int sila, int inicjatywa, char znak_gatunku, int wiek, string nazwa, Swiat& swiat)
	: Roslina(polozenieX, polozenieY, sila, inicjatywa, znak_gatunku, wiek, nazwa, swiat)
{
}

Guarana::~Guarana()
{
}

void Guarana::kolizja(Organizm* o)
{
	o->SetSila(o->GetSila() + 3);
	string px = to_string(polozeniex);
	string py = to_string(polozeniey);
	string p2x = to_string(o->GetPolozenieX());
	string p2y = to_string(o->GetPolozenieY());
	swiat.WypiszZdarzenie(o->GetNazwa() + "(" + p2x + "," + p2y + ") zjadl " + nazwa + "(" + px + "," + py + "), dostaje +3 sily");
	swiat.UsunOrganizm(polozeniex, polozeniey);
}

void Guarana::PostawDziecko(int x, int y)
{
	int szansa = rand() % RNG_GUARANA;
	if (szansa == 0)
	{
		swiat.WypiszZdarzenie(nazwa + "(" + to_string(polozeniex) + "," + to_string(polozeniey) + ") rozsiewa sie na (" + to_string(x) + "," + to_string(y) + ")");
		swiat.DodajOrganizm(new Guarana(x, y, swiat));
		swiat.gotoxy(40, 8);
		return;
	}
	swiat.WypiszZdarzenie(nazwa + "(" + to_string(polozeniex) + "," + to_string(polozeniey) + ") nie rozsiewa sie");
	swiat.gotoxy(40, 8);
}